govc/tags/category/ls.go

```diff
@@ -87,18 +92,45 @@ func (r getResult) Write(w io.Writer) error {
 	return nil
 }
 
+type getCategoryNameID []tags.CategoryInfo
+
+func (r getCategoryNameID) Write(w io.Writer) error {
+	for i := range r {
+		fmt.Fprintln(w, r[i])
+	}
+	return nil
+}
+
 func (cmd *ls) Run(ctx context.Context, f *flag.FlagSet) error {
 
 	return withClient(ctx, cmd.ClientFlag, func(c *tags.RestClient) error {
-		categories, err := c.ListCategories(ctx)
+
+		var result getResult
+		var categoryNameID getCategoryNameID
+		var err error
+
+		categoryNameID, err = c.ListCategoriesByName(ctx)
 		if err != nil {
 			return err
 		}
 
-		result := getResult(categories)
-		cmd.WriteResult(result)
-		return nil
-
+		switch {
+		case cmd.JSON && !cmd.id:
+			return cmd.WriteResult(categoryNameID)
```

Comment:
This is to make sure that if ```-i``` and ```-json``` are both set, ```ls``` commands will list JSON format IDs

Reply:
I thought `-json` would include both the `Name` and `ID`?